fe/fe-core/src/main/java/com/starrocks/catalog/Type.java

```diff
@@ -587,8 +589,9 @@ public boolean isOnlyMetricType() {
                 isScalarType(PrimitiveType.PERCENTILE);
     }
 
-    public boolean isPredicableType() {
-        return !isOnlyMetricType() && !(this instanceof ArrayType);
+    public boolean isPredicableType(Predicate predicate) {
+        return !isOnlyMetricType() &&
+                (!(this instanceof ArrayType) || predicate instanceof IsNullPredicate);
```

Comment:
I don't think Ast of Predicate should be introduced in Type's judgment. It is better to judge whether this logic is in predicateBaseAndCheck, because this logic itself is within the scope of semantic analysis. As for the old Analyzer, it's completely obsolete, don't care about it

Reply:
Ok, it is updated.